### PR TITLE
Report target-db-version and supported versions in assessment/analyze-schema reports

### PR DESCRIPTION
**Pull request by makalaaneesh**
Analyze Schema: 
image
image

Assessment: 
image
image

### CODE REVIEW

File: yb-voyager/cmd/analyzeSchema.go
Comment (priyanshi-yb) [inline yb-voyager/cmd/analyzeSchema.go:268]: please remove them before merging.

File: yb-voyager/cmd/assessMigrationCommand.go
Comment (priyanshi-yb) [inline yb-voyager/cmd/assessMigrationCommand.go:1006]: testing right?
please make sure to remove these.

File: yb-voyager/cmd/assessMigrationCommand.go
Comment (priyanshi-yb) [inline yb-voyager/cmd/assessMigrationCommand.go:954]: nit: we can use the `minVersionsFixedIn` to check if its set or not.

File: yb-voyager/cmd/assessMigrationCommand.go
Comment (priyanshi-yb) [inline yb-voyager/cmd/assessMigrationCommand.go:1108]: same here as above

File: yb-voyager/src/version/constants.go
Comment (priyanshi-yb) [inline yb-voyager/src/version/constants.go:37]: can't we use utils.ErrExit?

File: yb-voyager/src/utils/commonVariables.go
Comment (priyanshi-yb) [inline yb-voyager/src/utils/commonVariables.go:141]: we don't need `isFixedIn` here as we will get the filtered ones from `queryissue` pkg, right?

File: yb-voyager/src/version/constants.go
Comment (makalaaneesh) [reply yb-voyager/src/version/constants.go]: utils has some structs whose fields are of type `YBVersion`. so there needs to be a dependency on utils -> version. adding version->utils dependency causes circular dependency, so i thought we can just panic (As this is in the init function, it will definitely be caught in dev testing)

File: yb-voyager/src/utils/commonVariables.go
Comment (makalaaneesh) [reply yb-voyager/src/utils/commonVariables.go:141]: We still need because we need to include in the report the (higher) versions that it might be supported in. So, for both UnsupportedFeature, UnsupportedQueryConstruct, we'd need to store `MinimumVersionsFixedIn`.

File: yb-voyager/cmd/assessMigrationCommand.go
Comment (makalaaneesh) [reply yb-voyager/cmd/assessMigrationCommand.go:954]: Ah, so it would be wrong to check if `minVersionsFixedIn` is not nil. Consider the case where out of 2 issues, first one has `minVersionsFixedIn= nil` and second one has `minVersionsFixedIn as some not nil value`. This is a case where we want to catch and throw an error, because we want all issues in an unsupportedFeature to have the same `minVersionsFixedIn` (at least for now. Maybe when we change the report structure into a table, this might change). But if we just check if `minVersionsFixedIn` is nil or not, we will fail to catch this case.

File: yb-voyager/src/utils/commonVariables.go
Comment (priyanshi-yb) [reply yb-voyager/src/utils/commonVariables.go]: yeah, for map I got it, I was asking about the `IsFixedIn()` function for the struct, to check the construct is fixed or not.

File: yb-voyager/cmd/assessMigrationCommand.go
Comment (priyanshi-yb) [reply yb-voyager/cmd/assessMigrationCommand.go:954]: Ah okay got it.

File: yb-voyager/src/utils/commonVariables.go
Comment (makalaaneesh) [reply yb-voyager/src/utils/commonVariables.go]: I added it based on our previous discussion:
Ideally all filtering of issues should happen in queryissue pkg layer, so we should not need to filter here. But until all code is moved out, we should also filter in analyze-schema layer. 

Although now you're moving a majority of the code from analyze-schema into queryissues, there will still be some regex-based code in analyze-schema layer, I assume?

Comment (priyanshi-yb) [reply yb-voyager/src/utils/commonVariables.go]: > Although now you're moving a majority of the code from analyze-schema into queryissues, there will still be some regex-based code in analyze-schema layer, I assume?

yeah right

File: yb-voyager/src/ybversion/constants.go
Comment (sanyamsinghal) [inline yb-voyager/src/ybversion/constants.go:46]: were these two version added for testing ?

File: yb-voyager/src/issue/dml.go
Comment (makalaaneesh) [reply yb-voyager/src/issue/dml.go:52]: My bad, I thought I had cleaned up the PR. Will ping again

File: yb-voyager/src/utils/commonVariables.go
Comment (makalaaneesh) [reply yb-voyager/src/utils/commonVariables.go:111]: Initially I wanted to ignore xml, but I'm having second thoughts. I don't want to block this PR because of that. Can I take it up in a subsequent PR?

Comment (priyanshi-yb) [reply yb-voyager/src/utils/commonVariables.go:111]: Yes sure

**Comment by makalaaneesh** [comment]
https://jenkins.dev.yugabyte.com/blue/organizations/jenkins/users%2Fyb-voyager-testing%2Fyb-voyager-testing-pipeline/detail/yb-voyager-testing-pipeline/4125/pipeline

**Review by priyanshi-yb — APPROVED** [review]
LGTM

**Pull request merged**
